WebDevStudios/wd_s
# Feature/svgo updates

### DESCRIPTION ###
Adjusts our SVGO config file a bit.

I noticed on a project that some SVGs were displaying in unexpected ways after build processes were run.

This is what the SVG came out looking like:
![icon before](https://d.pr/free/i/aJCBj1+)

This is because the icon had some `fill="none"` properties on it which were being stripped by our config file, which was stripping _all_ instances of the `fill` properties. Retaining `fill="none"` is important for cases like this where there are some elements of an SVG which need to retain `fill="none"`.

After updating the config file and re-running the build, we get this:
![icon after](https://d.pr/free/i/5sIhMN+)

Now, the pieces that were using `fill="none"` retain that property so the more complex shapes and patterns aren't being thrown off.

This update also adds two new properties to our display/get SVG template tags:
- stroke
- stroke-width

This allows us to pass in the stroke color and the stroke-width values so we can set one or both of our values:
![](https://d.pr/free/i/E0uCZ0+)

A default color is used if no fill color is set, as seen in the first example, but this allows us to set stroke and fill values for our SVGs.

This, of course, will work better on SVGs that actually have strokes to begin with which none of our core SVGs have. This is why you'll see a little clipping around the edges in the example above.

### OTHER ###

### STEPS TO VERIFY ###
Checkout the branch, run `npm run build` and play around with fill and stroke properties in both CSS and in the display/get SVG template tags.

Add some SVGs with complex shapes and see how they react.

### DOCUMENTATION ###
Will this pull request require updating the wd_s [wiki](https://github.com/WebDevStudios/wd_s/wiki)?

Yes – once merged in, I'll update the details around our SVG template tags to include the new stroke attributes.
